packages/ui/src/components/SearchField/SearchField.tsx

```diff
@@ -79,13 +80,26 @@ export const SearchField = forwardRef<HTMLDivElement, SearchFieldProps>(
       props.onFocusChange?.(isFocused);
       if (shouldCollapse) {
         if (isFocused) {
-          setIsCollapsed(true);
-        } else {
+          // When focusing, expand the field
           setIsCollapsed(false);
+        } else {
+          // When blurring, collapse the field
+          setIsCollapsed(true);
         }
       }
     };
 
+    const handleContainerClick = () => {
+      // If the field is collapsed (small), expand it and focus the input
+      if (startCollapsed && isCollapsed) {
```

Comment:
Why should this logic only apply when you set `startCollapsed` to true? Don't you want this behavior when you started with the search field expanded?